### PR TITLE
initial commit for wavefront chart

### PR DESCRIPTION
**Pull request by puckpuck**
#### What this PR does / why we need it:

Initial commit for Wavefront monitoring

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by puckpuck** [comment]
What do I need to do to get this merged @cpanato and @prydonius ?  Just got the stale notice for this, not sure if something is required by me in order to get this merged in.

**Comment by cpanato** [comment]
sorry for the delay, i will look this chart tomorrow.

File: stable/wavefront/.helmignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+*.swp
+*.bak
+*.tmp
+*~
+# Various IDEs
+.project
+.idea/
+*.tmproj
+# OWNERS file for Kubernetes
+OWNERS
```
Comment (cpanato) [inline stable/wavefront/.helmignore]: new line :)

File: stable/wavefront/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+name: wavefront
+description: Wavefront Kubernetes collector
+appVersion: 0.9.6
+version: 0.2.0
```
Comment (cpanato) [inline stable/wavefront/Chart.yaml]:
```suggestion
version: 1.0.0
```

File: stable/wavefront/OWNERS [hunk 3]
```diff
@@ -0,0 +1,8 @@
+approvers:
+- vikramraman 
+- puckpuck
+- krisclarkdev
+reviewers:
+- vikramraman 
+- puckpuck
+- krisclarkdev
```
Comment (cpanato) [inline stable/wavefront/OWNERS]: new line

File: stable/wavefront/requirements.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+dependencies:
+  - name: kube-state-metrics
+    version: ^0.8.0
+    repository: https://kubernetes-charts.storage.googleapis.com/
+    condition: kubeStateMetrics.enabled
```
Comment (cpanato) [inline stable/wavefront/requirements.yaml]: new line

File: stable/wavefront/templates/api-token-secret.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    app.kubernetes.io/name : {{ template "wavefront.fullname" . }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io.instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/component: collector
+  name: {{ template "wavefront.fullname" . }}
+type: Opaque
+data:
+  api-token: {{ .Values.wavefront.token | b64enc | quote }}
+{{- end }}
```
Comment (cpanato) [inline stable/wavefront/templates/api-token-secret.yaml]: new line

File: stable/wavefront/templates/api-token-secret.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{{- if .Values.wavefront.token }}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name : {{ template "wavefront.fullname" . }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
```
Comment (cpanato) [inline stable/wavefront/templates/api-token-secret.yaml]: add the template in the helpers to use across all `helm.sh/chart` label 
```
{{/*
Create chart name and version as used by the chart label.
*/}}
{{- define "wavefront.chart" -}}
{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
{{- end -}}
```

File: stable/wavefront/templates/collector-config.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  labels:
+    app.kubernetes.io/name : {{ template "wavefront.fullname" . }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io.instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/component: collector    
+  name: {{ template "wavefront.collector.fullname" . }}-config
+data:
+{{ tpl (toYaml .Values.collector.discovery.config) . | indent 2 }}
+{{- end }}
```
Comment (cpanato) [inline stable/wavefront/templates/collector-config.yaml]: new line

File: stable/wavefront/templates/collector-deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io.instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/component: collector
+  name: {{ template "wavefront.collector.fullname" . }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name : {{ template "wavefront.fullname" .}}
+      app.kubernetes.io/component: collector
```
Comment (cpanato) [inline stable/wavefront/templates/collector-deployment.yaml:17]: add `app.kubernetes.io/instance`
follow this https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#deployments-statefulsets-daemonsets-selectors

File: stable/wavefront/templates/proxy-deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io.instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/component: proxy
+  name: {{ template "wavefront.proxy.fullname" . }} 
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name : {{ template "wavefront.fullname" .}}
+      app.kubernetes.io/component: proxy
```
Comment (cpanato) [inline stable/wavefront/templates/proxy-deployment.yaml:17]: same here

**Review by cpanato — CHANGES_REQUESTED** [review]

File: stable/wavefront/templates/collector-deployment.yaml (hunk 8 above)
Comment (puckpuck) [reply stable/wavefront/templates/collector-deployment.yaml:17]: I'm not quite clear on what I'm doing incorrectly here.  I do have the component added as a suffix for the name on this deployment.  For this chart that is done as a separate template in _helpers.tpl `wavefront.collector.fullname`. I did similar for the proxy component as well.

https://github.com/puckpuck/charts/blob/wavefront/stable/wavefront/templates/_helpers.tpl#L34-L48

Comment (puckpuck) [reply stable/wavefront/templates/collector-deployment.yaml:17]: @cpanato checking to see if the above explanation makes sense, and if not can you clarify what you need from me to finish this PR.

Thanks.

**Comment by puckpuck** [comment]
@cpanato and @prydonius This PR has been up for a while.  I addressed all the requested changes above except for 1 which I provided an explanation for.  We are very eager to get this merged into master.  Anything I can do to help expedite this?

**Comment by puckpuck** [comment]
@cpanato and @prydonius I'm not trying to be a pest.  Eager for some feedback and/or details on when we can get this PR approved and merged.

**Comment by cpanato** [comment]
@puckpuck my apologizes i've been busy with my work. i will review this

**Review by cpanato — APPROVED** [review]

**Pull request merged**